File: Back/Controllers/Dtos/QuizDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Ynov.QuizYnov.Controllers.Dtos
{
    public class QuizDto
    {
        public Guid Id { get; set; }

        public string? Name { get; set; }

        public string? Description { get; set; }

        public int Difficulty { get; set; }

        public DateTime PublishedDate { get; set; }

        public Guid CategoryId { get; set; }
    }
}

File: Front/src/app/business/services/dtos/quiz.dto.ts
import { QuestionsDto } from './question.dto';
import { CategoryDto } from './category.dto';

export interface QuizDto {
    readonly id: string;

    readonly name: string;

    readonly description: string;

    readonly difficulty: number;

    readonly publishedDate: Date;

    readonly categoryId: string;

    readonly category?: CategoryDto;

    readonly questions?: QuestionsDto[];
}
